x/launch/keeper/msg_create_chain_test.go

```diff
@@ -3,38 +3,129 @@ package keeper_test
 import (
 	"testing"
 
-	testkeeper "github.com/tendermint/spn/testutil/keeper"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
 	"github.com/stretchr/testify/require"
+	campaigntypes "github.com/tendermint/spn/x/campaign/types"
 
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
 	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/launch/keeper"
 	"github.com/tendermint/spn/x/launch/types"
 	profiletypes "github.com/tendermint/spn/x/profile/types"
 )
 
+func initCreationFeeAndFundCoordAccounts(
+	t *testing.T,
+	keeper *keeper.Keeper,
+	bk bankkeeper.Keeper,
+	sdkCtx sdk.Context,
+	fee sdk.Coins,
+	numCreations int64,
+	addrs ...string,
+) {
+	// set fee param to `coins`
+	params := keeper.GetParams(sdkCtx)
+	params.ChainCreationFee = fee
+	keeper.SetParams(sdkCtx, params)
+
+	coins := sdk.NewCoins()
+	for _, coin := range fee {
+		coin.Amount = coin.Amount.MulRaw(numCreations)
+		coins = coins.Add(coin)
+	}
+
+	// add `coins` to balance of each coordinator address
+	// using `campaign` module account for minting as `launch` does not have one
+	for _, addr := range addrs {
+		accAddr, err := sdk.AccAddressFromBech32(addr)
+		require.NoError(t, err)
+		err = bk.MintCoins(sdkCtx, campaigntypes.ModuleName, coins)
+		require.NoError(t, err)
+		err = bk.SendCoinsFromModuleToAccount(sdkCtx, campaigntypes.ModuleName, accAddr, coins)
+		require.NoError(t, err)
+	}
+}
+
 func TestMsgCreateChain(t *testing.T) {
-	sdkCtx, tk, ts := testkeeper.NewTestSetup(t)
-	ctx := sdk.WrapSDKContext(sdkCtx)
+	var (
+		coordAddr1       = sample.Address()
+		coordAddr2       = sample.Address()
+		coordAddr3       = sample.Address()
+		coordAddr4       = sample.Address()
+		coordAddr5       = sample.Address()
+		sdkCtx, tk, ts   = testkeeper.NewTestSetup(t)
+		ctx              = sdk.WrapSDKContext(sdkCtx)
+		chainCreationFee = sample.Coins()
+	)
 
 	// Create an invalid coordinator
 	invalidCoordAddress := sample.Address()
 	msgCreateInvalidCoordinator := sample.MsgCreateCoordinator(invalidCoordAddress)
 	_, err := ts.ProfileSrv.CreateCoordinator(ctx, &msgCreateInvalidCoordinator)
 	require.NoError(t, err)
 
-	// Create a coordinator
-	coordAddress := sample.Address()
-	msgCreateCoordinator := sample.MsgCreateCoordinator(coordAddress)
+	// Create coordinators
+	coordMap := make(map[string]uint64)
+	msgCreateCoordinator := sample.MsgCreateCoordinator(coordAddr1)
 	resCoord, err := ts.ProfileSrv.CreateCoordinator(ctx, &msgCreateCoordinator)
 	require.NoError(t, err)
-	coordID := resCoord.CoordinatorID
+	coordMap[coordAddr1] = resCoord.CoordinatorID
```

Comment:
Could we use a `for` loop for this?

Reply:
done the same also for `msg_create_campaign_test.go`. I didn't use a for loop here for consistency wrt to that file, but with a for loop for both it objectively looks cleaner. I've also switched to `sample.MsgCreateCoordinator()` while before it was using a custom declaration.